src/lib/UIComponent.tsx

```diff
@@ -1,7 +1,12 @@
-import React from 'react'
+import React, { HTMLAttributes } from 'react'
 import renderComponent, { IRenderResultConfig } from './renderComponent'
 
-class UIComponent<P, S> extends React.Component<P, S> {
+export interface UIComponentProps extends HTMLAttributes<HTMLElement> {
+  as?: string | Function
+  variables?: (siteVariables: object) => object
```

Comment:
actually, we should make it clear whether plain object could be accepted as value for `variables` prop - there are quite a few places where this approach is used now

Reply:
my vote would be to support both options, due to otherwise we could introduce unnecessary additional complexity for the client of this code in certain cases (where it is about just providing some predefined variables independent of theming aspects)

Reply:
I will address this, agreed.

Reply:
This will go away with #16, which adds typings for all theming concerns:

https://github.com/stardust-ui/react/blob/418e94761913b8bfc1035982d115693a406d0c12/types/theme.d.ts